chemprop/cli/train.py

```diff
@@ -526,33 +494,49 @@ def main(args):
             )
             train_data = all_data
         else:
-            if multicomponent:
-                train_data, val_data, _ = split_multicomponent(all_data, args.split, **split_kwargs)
-            else:
-                train_data, val_data, _ = split_monocomponent(all_data, args.split, **split_kwargs)
+            train_data, val_data, _ = split_fn(all_data, args.split, **split_kwargs)
     else:
         raise ArgumentError(
             argument=None, message="'val_path' must be specified if 'test_path' is provided!"
         )  # TODO: In v1 this wasn't the case?
 
     if multicomponent:
-        logger.info(
-            f"train/val/test sizes: {len(train_data[0])}/{len(val_data[0])}/{len(test_data[0])}"
-        )
+        sizes = [len(train_data[0]), len(val_data[0]), len(test_data[0])]
     else:
-        logger.info(f"train/val/test sizes: {len(train_data)}/{len(val_data)}/{len(test_data)}")
+        sizes = [len(train_data), len(val_data), len(test_data)]
+
+    logger.info(f"train/val/test sizes: {sizes}")
+
+    return train_data, val_data, test_data
```

Comment:
(I should also clarify that the signature of `build_data_from_files()` currently says it returns `list[MoleculeDatapoint]` which is incorrect because it calls `make_datapoints()` which returns `tuple[list[list[MoleculeDatapoint]], list[list[ReactionDatapoint]]]` and `build_data_from_files()` selects one of the double lists to return.)